packages/@sanity/form-builder/src/inputs/PortableText/object/BlockObject.tsx

```diff
@@ -0,0 +1,327 @@
+    onChange,
+    onFocus,
+    readOnly,
+    renderBlockActions,
+    renderCustomMarkers,
+    type,
+  } = props
+  const elementRef = useRef<HTMLDivElement>()
+
+  const handleEdit = useCallback(() => {
+    onFocus(path.concat(FOCUS_TERMINATOR))
+  }, [onFocus, path])
+
+  const handleClickToOpen = useCallback(() => {
+    handleEdit()
+  }, [handleEdit])
+
+  const handleDelete = useCallback(() => {
+    PortableTextEditor.delete(
+      editor,
+      {focus: {path, offset: 0}, anchor: {path, offset: 0}},
+      {mode: 'block'}
+    )
+    PortableTextEditor.focus(editor)
+  }, [editor, path])
+
+  const blockPreview = useMemo(() => {
+    return (
+      <BlockObjectPreview
+        type={type}
+        value={block}
+        readOnly={readOnly}
+        onClickingDelete={handleDelete}
+        onClickingEdit={handleEdit}
+      />
+    )
+  }, [type, block, readOnly, handleDelete, handleEdit])
+
+  const tone = useMemo(() => {
+    if (selected || focused) {
+      return 'primary'
+    }
+
+    return 'default'
+  }, [focused, selected])
+
+  const innerPaddingProps: ResponsivePaddingProps = useMemo(() => {
+    if (isFullscreen && !renderBlockActions) {
+      return {paddingX: 5}
+    }
+
+    if (isFullscreen && renderBlockActions) {
+      return {paddingLeft: 5, paddingRight: 2}
+    }
+
+    if (renderBlockActions) {
+      return {
+        paddingLeft: 3,
+        paddingRight: 2,
+      }
+    }
+
+    return {paddingX: 3}
+  }, [isFullscreen, renderBlockActions])
+
+  // These are marker that is only for the block level (things further up, like annotations and inline objects are dealt with in their respective components)
+  const blockMarkers = useMemo(
+    () =>
+      markers.filter(
+        (marker) => isKeySegment(marker.path[0]) && marker.path[0]._key === block._key
+      ),
+    [block._key, markers]
+  )
+
+  const errorMarkers = useMemo(
+    () => blockMarkers.filter((marker) => marker.type === 'validation' && marker.level === 'error'),
+    [blockMarkers]
+  )
+
+  const warningMarkers = useMemo(
+    () =>
+      blockMarkers.filter((marker) => marker.type === 'validation' && marker.level === 'warning'),
+    [blockMarkers]
+  )
+
+  const hasMarkers = blockMarkers.length > 0
+  const hasErrors = errorMarkers.length > 0
+  const hasWarnings = warningMarkers.length > 0
+
+  const markersToolTip = useMemo(
+    () =>
+      hasErrors || hasWarnings || (hasMarkers && renderCustomMarkers) ? (
+        <Tooltip
+          placement="top"
+          portal="editor"
+          content={
+            <TooltipBox padding={2}>
+              <Markers markers={markers} renderCustomMarkers={renderCustomMarkers} />
+            </TooltipBox>
+          }
+        >
+          <div>{blockPreview}</div>
+        </Tooltip>
+      ) : undefined,
+    [blockPreview, hasErrors, hasWarnings, hasMarkers, markers, renderCustomMarkers]
+  )
+
+  const isImagePreview = useMemo(() => type?.type?.name === 'image', [type?.type?.name])
```

Comment:
Overkill to use `useMemo`.
```suggestion
  const isImagePreview = type?.type?.name === 'image'
```